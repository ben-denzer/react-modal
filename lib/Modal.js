"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
exports.__esModule = true;
var React = require("react");
var ModalStyles_1 = require("./ModalStyles");
var Modal = /** @class */ (function (_super) {
    __extends(Modal, _super);
    function Modal(props) {
        var _this = _super.call(this, props) || this;
        _this.state = {
            buttonIsHovered: false,
            modalClassName: ''
        };
        _this.buttonMouseOut = _this.buttonMouseOut.bind(_this);
        _this.closeOnBgClick = _this.closeOnBgClick.bind(_this);
        _this.mouseOverButton = _this.mouseOverButton.bind(_this);
        _this.toggleModalClass = _this.toggleModalClass.bind(_this);
        return _this;
    }
    Modal.prototype.componentDidUpdate = function (prevProps) {
        var nextModalState = this.props.shouldShowModal;
        if (prevProps.shouldShowModal !== nextModalState) {
            this.toggleModalClass(nextModalState);
        }
    };
    Modal.prototype.buttonMouseOut = function () {
        this.setState({ buttonIsHovered: false });
    };
    Modal.prototype.closeOnBgClick = function (e) {
        var dataset = e.target.dataset;
        if (!dataset.targetId || dataset.targetId !== 'modalBg') {
            return;
        }
        this.props.closeModal();
    };
    Modal.prototype.calculateBgStyle = function () {
        var modalClassName = this.state.modalClassName;
        var styleMap = {
            open: ModalStyles_1.modalBackgroundOpen,
            transition: ModalStyles_1.modalBackgroundTransition
        };
        var defaultOverrides = styleMap[modalClassName] || {};
        return __assign({}, ModalStyles_1.modalBackground, defaultOverrides);
    };
    Modal.prototype.calculateButtonStyle = function () {
        var buttonIsHovered = this.state.buttonIsHovered;
        var hoverStyle = {
            cursor: 'pointer',
            color: '#222'
        };
        return buttonIsHovered ? __assign({}, ModalStyles_1.closeButton, hoverStyle) : ModalStyles_1.closeButton;
    };
    Modal.prototype.mouseOverButton = function () {
        this.setState({ buttonIsHovered: true });
    };
    Modal.prototype.toggleModalClass = function (shouldShowModal) {
        var _this = this;
        this.setState({ modalClassName: 'transition' });
        var minUpdateTime = 17;
        var nextClass = shouldShowModal ? 'open' : '';
        var timer = shouldShowModal ? minUpdateTime : ModalStyles_1.modalAnimationTime;
        setTimeout(function () { return _this.setState({ modalClassName: nextClass }); }, timer);
    };
    Modal.prototype.render = function () {
        return (React.createElement("div", { style: this.calculateBgStyle(), "data-target-id": "modalBg", onClick: this.closeOnBgClick },
            React.createElement("div", { style: ModalStyles_1.modalInner },
                React.createElement("div", { style: ModalStyles_1.modalHeader },
                    React.createElement("div", null, this.props.title),
                    React.createElement("div", { onClick: this.props.closeModal, onMouseOut: this.buttonMouseOut, onMouseOver: this.mouseOverButton, style: this.calculateButtonStyle() }, "X")),
                React.createElement("div", { style: ModalStyles_1.modalBody }, this.props.children))));
    };
    return Modal;
}(React.Component));
exports["default"] = Modal;
