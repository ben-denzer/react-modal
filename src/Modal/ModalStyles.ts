const headerHeight = 50;
const interiorPadding = 20;
export const modalAnimationTime = 300;

export const closeButton: React.CSSProperties = {
  height: `${headerHeight}px`,
  width: `${headerHeight}px`,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  fontWeight: 'normal',
  color: '#999'
};

export const modalBackground: React.CSSProperties = {
  position: 'fixed',
  zIndex: 9999,
  height: '100vh',
  width: '100vw',
  top: '0',
  left: '0',
  backgroundColor: 'rgba(0, 0, 0, 0.4)',
  display: 'none',
  transition: `all ${modalAnimationTime / 1000}s ease-in-out`
};

export const modalBackgroundTransition: React.CSSProperties = {
  display: 'block',
  opacity: 0,
  paddingTop: 0
};

export const modalBackgroundOpen: React.CSSProperties = {
  display: 'block',
  opacity: 1,
  paddingTop: '45px'
};

export const modalBody: React.CSSProperties = {
  padding: `${interiorPadding / 2}px ${interiorPadding}px`
};

export const modalHeader: React.CSSProperties = {
  display: 'flex',
  height: `${headerHeight}px`,
  justifyContent: 'space-between',
  alignItems: 'center',
  paddingLeft: `${interiorPadding}px`,
  fontSize: '18px',
  fontWeight: 'bold',
  borderBottom: '1px solid #ccc'
};

export const modalInner: React.CSSProperties = {
  margin: '0 auto',
  width: '98%',
  maxWidth: '700px',
  minHeight: '200px',
  backgroundColor: 'white',
  borderRadius: '2px',
  fontFamily: 'Arial, Helvetica, sans-serif'
};
