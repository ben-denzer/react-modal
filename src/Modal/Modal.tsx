import * as React from 'react';
import { ICustomModalStyle, IModalStyle, modalStyle } from './ModalStyles';

interface IModalProps {
  shouldShowModal: boolean;
  customStyle?: ICustomModalStyle;
  onlyCloseWithButton?: boolean;
  title?: string;
  closeModal(): void;
}

interface IModalState {
  modalAnimationTime: number;
  buttonIsHovered: boolean;
  modalClassName: '' | 'open' | 'transition';
}

class Modal extends React.Component<IModalProps, IModalState> {
  public static defaultProps: Partial<IModalProps> = {
    customStyle: {}
  };
  constructor(props: IModalProps) {
    super(props);
    this.state = {
      modalAnimationTime: this.setAnimationTime(),
      buttonIsHovered: false,
      modalClassName: ''
    };
    this.buttonMouseOut = this.buttonMouseOut.bind(this);
    this.closeOnBgClick = this.closeOnBgClick.bind(this);
    this.mouseOverButton = this.mouseOverButton.bind(this);
    this.setAnimationTime = this.setAnimationTime.bind(this);
    this.toggleModalClass = this.toggleModalClass.bind(this);
  }

  componentDidUpdate(prevProps: IModalProps): void {
    const nextModalState: boolean = this.props.shouldShowModal;
    if (prevProps.shouldShowModal !== nextModalState) {
      this.toggleModalClass(nextModalState);
    }
  }

  buttonMouseOut(): void {
    this.setState({ buttonIsHovered: false });
  }

  closeOnBgClick(e: React.MouseEvent<HTMLElement>): void {
    const { onlyCloseWithButton } = this.props;
    const { id } = e.target as HTMLElement;
    if (onlyCloseWithButton || !id || id !== 'modal-modalBackground') {
      return;
    }
    this.props.closeModal();
  }

  calculateBgStyle(
    modalBackground: React.CSSProperties,
    modalBackgroundOpen: React.CSSProperties,
    modalBackgroundTransition: React.CSSProperties
  ): React.CSSProperties {
    const { modalClassName } = this.state;
    const styleMap = {
      open: modalBackgroundOpen,
      transition: modalBackgroundTransition
    };
    const currentState = styleMap[modalClassName] || {};
    return { ...modalBackground, ...currentState };
  }

  calculateButtonStyle(
    closeButton: React.CSSProperties,
    closeButtonHover: React.CSSProperties,
    closeButtonText: React.CSSProperties,
    hoveredButtonText: React.CSSProperties
  ): { button: React.CSSProperties; buttonText: React.CSSProperties } {
    const { buttonIsHovered } = this.state;
    const button = buttonIsHovered
      ? { ...closeButton, ...closeButtonHover }
      : closeButton;
    const buttonText = buttonIsHovered
      ? { ...closeButtonText, ...hoveredButtonText }
      : closeButtonText;
    return { button, buttonText };
  }

  mergeStyles(): IModalStyle {
    const custom = this.props.customStyle as ICustomModalStyle;
    const merged = { ...modalStyle };
    for (const i in custom) {
      if (custom.hasOwnProperty(i)) {
        merged[i] = { ...merged[i], ...custom[i] };
      }
    }

    // set animation time
    merged.modalBackground.transitionDuration = `${this.state
      .modalAnimationTime / 1000}s`;
    return merged;
  }

  mouseOverButton(): void {
    this.setState({ buttonIsHovered: true });
  }

  setAnimationTime(): number {
    const { customStyle } = this.props;
    if (customStyle) {
      if (customStyle.animationTime === 0 || customStyle.animationTime) {
        return customStyle.animationTime;
      }
    }
    return modalStyle.animationTime;
  }

  toggleModalClass(shouldShowModal: boolean): void {
    this.setState({ modalClassName: 'transition' });
    const { modalAnimationTime } = this.state;
    const minUpdateTime = 17;
    const nextClass = shouldShowModal ? 'open' : '';
    const timer: number = shouldShowModal ? minUpdateTime : modalAnimationTime;
    setTimeout(() => this.setState({ modalClassName: nextClass }), timer);
  }

  render(): JSX.Element {
    const mergedStyles = this.mergeStyles();
    const {
      closeButton,
      closeButtonText,
      closeButtonHover,
      hoveredButtonText,
      modalBackground,
      modalBackgroundTransition,
      modalBackgroundOpen,
      modalBody,
      modalHeader,
      modalInner,
      modalTitle
    } = mergedStyles;
    const bgStyle = this.calculateBgStyle(
      modalBackground,
      modalBackgroundOpen,
      modalBackgroundTransition
    );
    const { button, buttonText } = this.calculateButtonStyle(
      closeButton,
      closeButtonHover,
      closeButtonText,
      hoveredButtonText
    );

    return (
      <div
        style={bgStyle}
        id="modal-modalBackground"
        onClick={this.closeOnBgClick}
      >
        <div id="modal-modalInner" style={modalInner}>
          <div id="modal-modalHeader" style={modalHeader}>
            <div id="modal-modalTitle" style={modalTitle}>
              {this.props.title}
            </div>
            <div
              id="modal-closeButton"
              onClick={this.props.closeModal}
              onMouseOut={this.buttonMouseOut}
              onMouseOver={this.mouseOverButton}
              style={button}
            >
              <span style={buttonText} id="modal-closeButtonText">
                X
              </span>
            </div>
          </div>
          <div id="modal-modalBody" style={modalBody}>
            {this.props.children}
          </div>
        </div>
      </div>
    );
  }
}

export { ICustomModalStyle };
export default Modal;
