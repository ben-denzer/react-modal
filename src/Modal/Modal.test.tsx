import * as React from 'react';
import { mount, shallow } from 'enzyme';
import Modal from './Modal';

describe('Modal', () => {
  let closeModal: any;
  beforeEach(() => {
    closeModal = jest.fn();
  });

  it('should render without crashing', () => {
    shallow(<Modal closeModal={closeModal} shouldShowModal={true} />);
  });

  it('should show the title if it is given one', () => {
    const wrapper = shallow(
      <Modal closeModal={closeModal} shouldShowModal={true} title="My Title" />
    );
    const titleText = wrapper.find('#modal-modalTitle').text();
    expect(titleText).toEqual('My Title');
  });

  it('should show body content', () => {
    const wrapper = shallow(
      <Modal closeModal={closeModal} shouldShowModal={true}>
        Here is some body content
      </Modal>
    );
    const bodyText = wrapper.find('#modal-modalBody').text();
    expect(bodyText).toEqual('Here is some body content');
  });

  it('should call closeModal if close button is clicked', () => {
    const wrapper = shallow(
      <Modal closeModal={closeModal} shouldShowModal={true} />
    );
    wrapper.find('#modal-closeButton').simulate('click');
    expect(closeModal).toHaveBeenCalledTimes(1);
  });

  it('should call closeModal if background is clicked', () => {
    const wrapper = mount(
      <Modal closeModal={closeModal} shouldShowModal={true} />
    );
    wrapper.find('#modal-modalBg').simulate('click');
    expect(closeModal).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(
      <Modal closeModal={closeModal} shouldShowModal={true} title="My Title" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
