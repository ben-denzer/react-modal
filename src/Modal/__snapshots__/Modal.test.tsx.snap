// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    closeModal={[MockFunction]}
    customStyle={Object {}}
    shouldShowModal={true}
    title="My Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        id="modal-modalInner"
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": "2px",
            "display": "flex",
            "flexDirection": "column",
            "fontFamily": "Arial, Helvetica, sans-serif",
            "margin": "0 auto 200px",
            "maxWidth": "700px",
            "minHeight": "200px",
            "width": "98%",
          }
        }
      >
        <div
          id="modal-modalHeader"
          style={
            Object {
              "alignItems": "center",
              "borderBottom": "1px solid #ccc",
              "display": "flex",
              "fontSize": "18px",
              "fontWeight": "bold",
              "justifyContent": "space-between",
              "minHeight": "50px",
              "paddingLeft": "20px",
            }
          }
        >
          <div
            id="modal-modalTitle"
            style={Object {}}
          >
            My Title
          </div>
          <div
            id="modal-closeButton"
            onClick={[MockFunction]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "alignItems": "center",
                "color": "#999",
                "display": "flex",
                "fontWeight": "normal",
                "height": "50px",
                "justifyContent": "center",
                "width": "50px",
              }
            }
          >
            <span
              id="modal-closeButtonText"
              style={Object {}}
            >
              X
            </span>
          </div>
        </div>
        <div
          id="modal-modalBody"
          style={
            Object {
              "flexGrow": 1,
              "padding": "10px 20px",
            }
          }
        />
      </div>,
      "id": "modal-modalBackground",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "rgba(0, 0, 0, 0.4)",
        "display": "none",
        "height": "100vh",
        "left": "0",
        "overflow": "auto",
        "position": "fixed",
        "top": "0",
        "transitionDuration": "0.3s",
        "transitionProperty": "all",
        "transitionTimingFunction": "ease-in-out",
        "width": "100vw",
        "zIndex": 9999,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="modal-modalHeader"
            style={
              Object {
                "alignItems": "center",
                "borderBottom": "1px solid #ccc",
                "display": "flex",
                "fontSize": "18px",
                "fontWeight": "bold",
                "justifyContent": "space-between",
                "minHeight": "50px",
                "paddingLeft": "20px",
              }
            }
          >
            <div
              id="modal-modalTitle"
              style={Object {}}
            >
              My Title
            </div>
            <div
              id="modal-closeButton"
              onClick={[MockFunction]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "color": "#999",
                  "display": "flex",
                  "fontWeight": "normal",
                  "height": "50px",
                  "justifyContent": "center",
                  "width": "50px",
                }
              }
            >
              <span
                id="modal-closeButtonText"
                style={Object {}}
              >
                X
              </span>
            </div>
          </div>,
          <div
            id="modal-modalBody"
            style={
              Object {
                "flexGrow": 1,
                "padding": "10px 20px",
              }
            }
          />,
        ],
        "id": "modal-modalInner",
        "style": Object {
          "backgroundColor": "white",
          "borderRadius": "2px",
          "display": "flex",
          "flexDirection": "column",
          "fontFamily": "Arial, Helvetica, sans-serif",
          "margin": "0 auto 200px",
          "maxWidth": "700px",
          "minHeight": "200px",
          "width": "98%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="modal-modalTitle"
                style={Object {}}
              >
                My Title
              </div>,
              <div
                id="modal-closeButton"
                onClick={[MockFunction]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "color": "#999",
                    "display": "flex",
                    "fontWeight": "normal",
                    "height": "50px",
                    "justifyContent": "center",
                    "width": "50px",
                  }
                }
              >
                <span
                  id="modal-closeButtonText"
                  style={Object {}}
                >
                  X
                </span>
              </div>,
            ],
            "id": "modal-modalHeader",
            "style": Object {
              "alignItems": "center",
              "borderBottom": "1px solid #ccc",
              "display": "flex",
              "fontSize": "18px",
              "fontWeight": "bold",
              "justifyContent": "space-between",
              "minHeight": "50px",
              "paddingLeft": "20px",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My Title",
                "id": "modal-modalTitle",
                "style": Object {},
              },
              "ref": null,
              "rendered": "My Title",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  id="modal-closeButtonText"
                  style={Object {}}
                >
                  X
                </span>,
                "id": "modal-closeButton",
                "onClick": [MockFunction],
                "onMouseOut": [Function],
                "onMouseOver": [Function],
                "style": Object {
                  "alignItems": "center",
                  "color": "#999",
                  "display": "flex",
                  "fontWeight": "normal",
                  "height": "50px",
                  "justifyContent": "center",
                  "width": "50px",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "X",
                  "id": "modal-closeButtonText",
                  "style": Object {},
                },
                "ref": null,
                "rendered": "X",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "id": "modal-modalBody",
            "style": Object {
              "flexGrow": 1,
              "padding": "10px 20px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="modal-modalInner"
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": "2px",
              "display": "flex",
              "flexDirection": "column",
              "fontFamily": "Arial, Helvetica, sans-serif",
              "margin": "0 auto 200px",
              "maxWidth": "700px",
              "minHeight": "200px",
              "width": "98%",
            }
          }
        >
          <div
            id="modal-modalHeader"
            style={
              Object {
                "alignItems": "center",
                "borderBottom": "1px solid #ccc",
                "display": "flex",
                "fontSize": "18px",
                "fontWeight": "bold",
                "justifyContent": "space-between",
                "minHeight": "50px",
                "paddingLeft": "20px",
              }
            }
          >
            <div
              id="modal-modalTitle"
              style={Object {}}
            >
              My Title
            </div>
            <div
              id="modal-closeButton"
              onClick={[MockFunction]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "color": "#999",
                  "display": "flex",
                  "fontWeight": "normal",
                  "height": "50px",
                  "justifyContent": "center",
                  "width": "50px",
                }
              }
            >
              <span
                id="modal-closeButtonText"
                style={Object {}}
              >
                X
              </span>
            </div>
          </div>
          <div
            id="modal-modalBody"
            style={
              Object {
                "flexGrow": 1,
                "padding": "10px 20px",
              }
            }
          />
        </div>,
        "id": "modal-modalBackground",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "rgba(0, 0, 0, 0.4)",
          "display": "none",
          "height": "100vh",
          "left": "0",
          "overflow": "auto",
          "position": "fixed",
          "top": "0",
          "transitionDuration": "0.3s",
          "transitionProperty": "all",
          "transitionTimingFunction": "ease-in-out",
          "width": "100vw",
          "zIndex": 9999,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="modal-modalHeader"
              style={
                Object {
                  "alignItems": "center",
                  "borderBottom": "1px solid #ccc",
                  "display": "flex",
                  "fontSize": "18px",
                  "fontWeight": "bold",
                  "justifyContent": "space-between",
                  "minHeight": "50px",
                  "paddingLeft": "20px",
                }
              }
            >
              <div
                id="modal-modalTitle"
                style={Object {}}
              >
                My Title
              </div>
              <div
                id="modal-closeButton"
                onClick={[MockFunction]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "color": "#999",
                    "display": "flex",
                    "fontWeight": "normal",
                    "height": "50px",
                    "justifyContent": "center",
                    "width": "50px",
                  }
                }
              >
                <span
                  id="modal-closeButtonText"
                  style={Object {}}
                >
                  X
                </span>
              </div>
            </div>,
            <div
              id="modal-modalBody"
              style={
                Object {
                  "flexGrow": 1,
                  "padding": "10px 20px",
                }
              }
            />,
          ],
          "id": "modal-modalInner",
          "style": Object {
            "backgroundColor": "white",
            "borderRadius": "2px",
            "display": "flex",
            "flexDirection": "column",
            "fontFamily": "Arial, Helvetica, sans-serif",
            "margin": "0 auto 200px",
            "maxWidth": "700px",
            "minHeight": "200px",
            "width": "98%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  id="modal-modalTitle"
                  style={Object {}}
                >
                  My Title
                </div>,
                <div
                  id="modal-closeButton"
                  onClick={[MockFunction]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "color": "#999",
                      "display": "flex",
                      "fontWeight": "normal",
                      "height": "50px",
                      "justifyContent": "center",
                      "width": "50px",
                    }
                  }
                >
                  <span
                    id="modal-closeButtonText"
                    style={Object {}}
                  >
                    X
                  </span>
                </div>,
              ],
              "id": "modal-modalHeader",
              "style": Object {
                "alignItems": "center",
                "borderBottom": "1px solid #ccc",
                "display": "flex",
                "fontSize": "18px",
                "fontWeight": "bold",
                "justifyContent": "space-between",
                "minHeight": "50px",
                "paddingLeft": "20px",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "My Title",
                  "id": "modal-modalTitle",
                  "style": Object {},
                },
                "ref": null,
                "rendered": "My Title",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    id="modal-closeButtonText"
                    style={Object {}}
                  >
                    X
                  </span>,
                  "id": "modal-closeButton",
                  "onClick": [MockFunction],
                  "onMouseOut": [Function],
                  "onMouseOver": [Function],
                  "style": Object {
                    "alignItems": "center",
                    "color": "#999",
                    "display": "flex",
                    "fontWeight": "normal",
                    "height": "50px",
                    "justifyContent": "center",
                    "width": "50px",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "X",
                    "id": "modal-closeButtonText",
                    "style": Object {},
                  },
                  "ref": null,
                  "rendered": "X",
                  "type": "span",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "id": "modal-modalBody",
              "style": Object {
                "flexGrow": 1,
                "padding": "10px 20px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
